src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -504,6 +506,22 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			this._tableContainer.classList.remove('hide-username');
 			this._tableContainer.classList.add('hide-password');
 			this._tableContainer.classList.remove('hide-azure-accounts');
+		} else if (currentAuthType === AuthenticationType.dSTSAuth) {
+			this._accountManagementService.getAccountsForProvider('dstsAuth').then(accounts => {
+				if (accounts && accounts.length > 0) {
+					accounts[0].key.providerArgs = {
+						serverName: this.serverName,
+						databaseName: this.databaseName
+					};
+
+					this._accountManagementService.getAccountSecurityToken(accounts[0], '', null).then(securityToken => {
+						this._token = securityToken.token;
+					});
+				}
+			});
+			this._tableContainer.classList.add('hide-username');
```

Comment:
I don't see where these styles get removed when the user changes back to SqlLogin auth.  But I assume that happens somewhere.